### PR TITLE
CSHARP-4808: "Updating stale electionId and setVersion" even when electionId/setVersion match

### PR DESCRIPTION
**Pull request by BorisDog**

### CODE REVIEW

File: src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs [hunk 1]
```diff
@@ -707,7 +707,9 @@ public override bool Equals(object obj)
             {
                 var other = obj as ElectionInfo;
 
-                return other != null && _setVersion == other.SetVersion && other.ElectionId == _electionId;
+                return other != null &&
+                    _setVersion == other.SetVersion &&
+                    (_electionId == null && other.ElectionId == null || other.ElectionId?.Equals(_electionId) == true);
```
Comment (rstam) [inline src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs:712]: I think you can simplify this line to:

    object.Equals(_electionId, other.ElectionId)

**Review by rstam — CHANGES_REQUESTED** [review]
One minor change suggested.

File: src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs (hunk 1 above)
Comment (BorisDog) [reply src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs]: Good suggestion, thanks!

**Review by rstam — APPROVED** [review]
LGTM

**Pull request merged**